### ISSUE TITLE
properly handle relative linking in Markdown by Flow documentation build system

### ISSUE THREAD

**Issue by kerrywei** [onflow/flow#704]
# context
[Flow documentation site](https://docs.onflow.org/) pulls contents from other Github repos so that the documentation of these tools can be displayed in the official doc site. 
Suppose I have two Markdown files: index.md and tutorial1.md, and I want to link some word in index.md to some section in tutorial1.md, I'll use relative path when linking in order to get the documentation works in my Github repo (or any documentation system I use): 
[some_word](tutorial1.md#section1)
However, some pages in the Flow official documentation site has trailing / appened. If my page is assigned URL docs.onflow.org/my-tool/index/ , the relative link in this page will be pointing to docs.onflow.org/my-tool/index/tutorial1#section1, which is a 404 page

# suggestions
When converting links in Markdown to fit into Flow official web pages, ensure the Markdown relative linking works properly

**Comment by kerrywei** [comment]
update: 
found one case that we probably want to handle:
Suppose in flow-js-testing library, there have two files: template.md and api.md, and I want to make a reference in template.md:
`[API Response](./api.md#response)`

if I include the file suffix .md, the link works in github but NOT in flow doc site as there is no https://docs.onflow.org/flow-js-testing/api.md#response (because our URL path do not include suffix);
if I ignore the file suffix ,md, the link works in flow doc site, but does not work in github

Surfaced this to Mac, meanwhile I'll prepare changes in other repos and coordinate doc release after this suffix is properly handled

**Comment by 10thfloor** [comment]
Since this issue is taking longer than expected, I'll add some notes on what we have done to address it. 

**Testing current patterns**
Discovery issue: The Flow JS Testing repo

## Objective of this issue
To. provide a consistent authoring experience for devs, specifically creating relative links to other documents in the content hierarchy. 

## **Understand the issue**

### Gatsby and trailing slashes
1) Gatsby builds pages for markdown files, and creates a URL for each page that looks like this, by default: 
 - A page located at (content root) `/node-operation.md` is available at `docs.onflow.org/node-operation/`
 - A page in a subdirectory at (content root) `/emulator/install.md` is available at `docs.onflow.org/emulator/install/`
 - Any page named _`index.md`_ will serve as a "root page" for a folder, for example: 
     - (content root) `/emulator/index.md` is available at `docs.onflow.org/emulator`
     
* The important detail is: non-`index.md` pages are served by default with a *trailing slash* while `index.md` pages  are served _without_ a trailing slash. 

### Community Best Practice and Discussion
The reasons for this default have to do with webservers, and are discussed in detail here: https://github.com/gatsbyjs/gatsby/discussions/27889, and here: https://jonsully.net/blog/trailing-slashes-and-gatsby/

tl;dr For our deployment. we will use this behaviour.

### Issues with this behaviour

Given this behaviour, relative links between documents are different depending on if they are `index.md` or other non-index pages. For index documents, relative links to sibling documents, for example, must be in the form `./` while. non-index pages must be `../` because of the trailing slash! 

As we'll see, this requirement is not consistent with content authored in Github Repositories. 

Github to docs site github process
Docsite deployment
Where does the liink transformation happen?

Looking for existing solutions 
Gatsby flow rewrite links plugin

Porting existing solutions for our use case
Creating a regex. to capture happy path

More testing and edge cases
Trailing slashes applied but not respected

**Comment by 10thfloor** [comment]
Given the preceding setup, here are the rules we must follow when authoring, or adding links to _docs site content only_.
- All uses of Gatsby’s <Link> component in your code to intentionally specify a trailing slash.

This means anywhere you're constructing a link manually, you must include the trailing slash. This only applies to content being added via the docs site hence not being passed through the link transformation code above.
